stock_request/models/stock_request.py

```diff
@@ -259,6 +259,10 @@ def action_request(self):
         self.ensure_one()
         if not self.line_ids:
             raise ValidationError(_('No lines!'))
+        for line in self.line_ids:
```

Comment:
```suggestion
        product_empty = self.line_ids.filtered(lambda l: l.product_id.qty_available == 0.0)
        if product:
            raise Valida....
```
filtered is better than for loop